### PR TITLE
[Heartbeat] Fix source conflict, improve project naming

### PR DESCRIPTION
**Pull request by andrewvc**
Fixes bug introduced in #31688  where new `source` field conflicts with browser type's `source` field. Renamed `source` to origin in stdfields.

This also cleans up more references to suites moving them to project, and creates a proper mapping for project fields, moving them into the `monitor.project` key which is cleaner than the prior unmapped top-level `project` key.

No changelog needed because there are no user-facing changes.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: heartbeat/config/config.go [hunk 1]
```diff
@@ -27,13 +27,12 @@ import (
 
 // Config defines the structure of heartbeat.yml.
 type Config struct {
-	RunOnce         bool                 `config:"run_once"`
-	Monitors        []*conf.C            `config:"monitors"`
-	ConfigMonitors  *conf.C              `config:"config.monitors"`
-	Scheduler       Scheduler            `config:"scheduler"`
-	Autodiscover    *autodiscover.Config `config:"autodiscover"`
-	SyntheticSuites []*conf.C            `config:"synthetic_suites"`
```
Comment (andrewvc) [inline heartbeat/config/config.go:35]: Removed this line, this is from some very early alphas where we had a separate config entry

File: x-pack/heartbeat/monitors/browser/suite.go [hunk 2]
```diff
@@ -18,19 +18,19 @@ import (
 	"github.com/elastic/elastic-agent-libs/mapstr"
```
Comment (vigneshshanmugam) [inline x-pack/heartbeat/monitors/browser/suite.go:18]: nit: should we rename this file to project.go?

File: x-pack/heartbeat/monitors/browser/suite.go [hunk 3]
```diff
@@ -39,73 +39,73 @@ func NewSuite(rawCfg *config.C) (*Suite, error) {
 }
 
 func ErrBadConfig(err error) error {
-	return fmt.Errorf("could not parse suite config: %w", err)
+	return fmt.Errorf("could not parse project config: %w", err)
 }
 
-func (s *Suite) String() string {
+func (s *Project) String() string {
```
Comment (vigneshshanmugam) [inline x-pack/heartbeat/monitors/browser/suite.go:45]: nit: `func (p *Project)`

**Review by vigneshshanmugam — APPROVED** [review]
LGTM

**Comment by justinkambic** [comment]
Post-FF Testing LGTM

To test this I configured a zip URL monitor in my `heartbeat.yml` like:

```yaml
- type: browser
  id: todos-project
  name: Todos Journeys
  enabled: true
  schedule: '@every 1m'
  source:
    zip_url:
      url: "https://github.com/elastic/synthetics/archive/refs/heads/master.zip"
      folder: "examples/todos"
      username: ""
      password: ""
```

I observed the `project` field as described in this PR in the documents my agent was logging in ES:

```JSON
{
  "monitor": {
    "check_group": "e2b56fd0-e69d-11ec-b543-8a2db4a65dcf",
    "id": "todos-project-addition and completion of single task",
    "name": "Todos Journeys - addition and completion of single task",
    "type": "browser",
    "project": { // <-- project field
      "id": "todos-project",
      "name": "Todos Journeys"
    }
  }
}
```

**Pull request merged**